app/tasksmodule/datachecks.py

```diff
@@ -0,0 +1,178 @@
+        q = model.Person.all(filter_expired=False).filter(
+            'repo =', self.env.repo)
+        cursor = self.params.get('cursor', '')
+        if cursor:
+            q.with_cursor(cursor)
+        try:
+            for person in q:
+                next_cursor = q.cursor()
+                self._check_person(person)
+                cursor = next_cursor
+        except runtime.DeadlineExceededError:
+            self._schedule_task(self.env.repo, cursor)
+        except datastore_errors.Timeout:
+            self._schedule_task(self.env.repo, cursor)
+        return django.http.HttpResponse('')
+
+
+class NoteDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that notes have valid data."""
+
+    BASE_NAME = 'note_data_validity_check'
+    TASK_PATH = 'check_note_data_validity'
+
+    def _check_note(self, note):
+        if not note.entry_date:
+            self.alert('A note record is missing an entry_date value.')
+        if not note.original_creation_date:
+            self.alert(
+                'A note record is missing an original_creation_date value.')
+        if not note.person_record_id:
+            self.alert('A note record is missing a person_record_id value.')
+        if (note.author_email and not utils.validate_email(note.author_email)):
+            self.alert('A note record has an invalid author_email value.')
+        if (note.email_of_found_person and
+                not utils.validate_email(note.email_of_found_person)):
+            self.alert(
+                'A note record has an invalid email_of_found_person value.')
+        if not model.Person.get(self.env.repo, note.person_record_id):
+            self.alert('A note record\'s associated person record is missing.')
+
+    def post(self, request, *args, **kwargs):
+        del request, args, kwargs  # unused
+        q = model.Note.all(filter_expired=False).filter(
+            'repo =', self.env.repo)
+        cursor = self.params.get('cursor', '')
+        if cursor:
+            q.with_cursor(cursor)
+        try:
+            for note in q:
+                next_cursor = q.cursor()
+                self._check_note(note)
+                cursor = next_cursor
+        except runtime.DeadlineExceededError:
+            self._schedule_task(self.env.repo, cursor)
+        except datastore_errors.Timeout:
+            self._schedule_task(self.env.repo, cursor)
+        return django.http.HttpResponse('')
+
+
+class ExpiredPersonRecordCheckTask(DatachecksBaseTask):
+    """Checks that expired person records have been cleared."""
+
+    BASE_NAME = 'expired_person_record_check'
+    TASK_PATH = 'check_expired_person_records'
+
+    def _check_person(self, person):
+        # Check things that were expired yesterday, just in case this job is
+        # ahead of the deletion job.
+        yesterday = utils.get_utcnow() - datetime.timedelta(days=1)
+        if person.expiry_date and person.expiry_date < yesterday:
```

Comment:
in case you wanted to reduce the nesting levels, you can turn that if block into an early return instead

Reply:
Oh, yeah, I surely do want to reduce the nesting.